### ISSUE TITLE
Test suite fails with -Duselongdouble

### ISSUE THREAD

**Issue by eserte** [DrHyde/perl-modules-Data-CompactReadonly#1]
If perl was compiled with increased floating point accuracy (using the config option `-Duselongdouble`), then the test suite fails:
```
PERL_DL_NONLAZY=1 "/usr/perl5.28.1Dp/bin/perl5.28.1" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t

#   Failed test 'read a Float'
#   at t/create-collection.t line 49.
#          got: '3.39999999999999991'
#     expected: '3.4'

#   Failed test 'read a Float'
#   at t/create-collection.t line 72.
#          got: '3.39999999999999991'
#     expected: '3.4'

#   Failed test 'read a Float'
#   at t/create-collection.t line 128.
#          got: '3.14000000000000012'
#     expected: '3.14'
Data::CompactReadonly::V0::Dictionary::Medium=HASH(0x80266de58): Invalid element: 0.07: doesn't exist
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 78.
t/create-collection.t ..... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 3/78 subtests 
... (etc, more tests are failing) ...
```

**Comment by DrHyde** [comment]
Thanks! I'd seen the test failure reports but you've saved me the effort of figuring out what was wrong :-)

**Comment by DrHyde** [comment]
Fixed in 5cfbb45f.

For my actual use case I don't need stupid floating point numbers at all and I wish I'd not been such a completist!
